server/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -1111,4 +1107,42 @@ public void test_update_object_array_by_elements_with_column_policies() {
             .isExactlyInstanceOf(ColumnUnknownException.class)
             .hasMessage("Column a['c'] unknown");
     }
+
+    @Test
+    public void test_update_on_table_with_big_refresh_interval_emptied_by_delete_return_0_rows() {
+        execute("create table test (a int) with (refresh_interval = 10000)");
+        execute("insert into test (a) values (1)");
+        assertThat(response).hasRowCount(1);
+        refresh();
+
+        execute("delete from test");
+        assertThat(response).hasRowCount(1);
+
+        // No refresh() here, collect doesn't now that table is already empty,
+        // which leads to ShardUpsertRequest to be issued.
+        // We ensure that no exception is thrown and UPDATE behaves similar to DELETE, returns 0 rows.
+        execute("update test set a = 2");
+        assertThat(response).hasRowCount(0L);
+
+        // Another similar scenario of update, which could hit multiple rows.
+        execute("update test set a = 2 where a = 1");
+        assertThat(response).hasRowCount(0L);
+    }
+
+    @Test
+    public void test_update_continues_on_error_if_at_least_1_row_was_updated() {
+        execute("create table test (a int CHECK (a < 100))");
+
+        execute("insert into test (a) values (1), (2), (3)");
+        assertThat(response).hasRowCount(3);
+        refresh();
+
+        // Trying to update 1, 2, 3 to (1 + 100), (2 + 0), (3 + 100), only (2 + 0) doesn't violate CHECK.
+        execute("update test set a = a + 100 * (a % 2)");
+        assertThat(response).hasRowCount(1); // We cannot rollback and 1 success prevents us from showing an error.
+
+        // TODO: restore old behavior, throw an exception if no row was updated and UPDATE hit > 0 rows.
+        execute("update test set a = a + 100");
```

Comment:
Same here, will remove TODO if it's fine and we _don't_ need to raise an error if `succefullRowsCount == 0`